IntelliTect.TestTools.TestFramework/IntelliTect.TestTools.TestFramework/TestBuilder.cs

```diff
@@ -222,7 +182,150 @@ private string GetObjectDataAsJsonString(object obj)
             }
         }
 
+        private object GetTestBlock(IServiceScope scope, Type tbType, ILogger logger)
+        {
+            logger.Info(TestCaseName, tbType.ToString(), "Starting test block.");
+
+            try
+            {
+                return scope.ServiceProvider.GetRequiredService(tbType);
+            }
+            catch (InvalidOperationException ex)
+            {
+                // Probably worth moving these logs outside of the foreach so we don't have to duplicate the message
+                logger.Error(TestCaseName, tbType.ToString(), "Unable to find the test block instance OR all dependencies necessary.");
+                TestBlockException = ex;
+                return null;
+            }
+
+        }
+
+        private void SetTestBlockProperties(IServiceScope scope, ref object testBlockInstance, ILogger logger)
```

Comment:
Is there a reason `testBlockInstance` is passed by ref?

Reply:
I wanted to make it super duper clear that the object being passed in was getting modified, even though it wasn't technically necessary. I suppose re-assigning it from the method return (or the implicitness based on the method name?) is a more appropriate way to do it, now that I look back on it?

Reply:
If the method is going to mutate the object, I would say you should be a `void` return and just pass the object normally. There is no way to really indicate this behavior back to the caller beyond the method name. Given that the name is `SetTestBlockProperties` I think it is perfectly clear that this method will be setting properties on the test block that I pass in. In addition this is a private method so the only people who need to understand this behavior are people who work on this class.

Reply:
Fair enough, will do (as soon as I get a second :) )!

Reply:
Ref removed with next commit